src/cargo/core/compiler/job_queue/mod.rs

```diff
@@ -826,6 +830,9 @@ impl<'gctx> DrainState<'gctx> {
             // `display_error` inside `handle_error`.
             Some(anyhow::Error::new(AlreadyPrintedError::new(error)))
         } else if self.queue.is_empty() && self.pending_queue.is_empty() {
+            build_runner.compilation.warning_count +=
+                self.warning_count.values().map(|c| c.total).sum::<usize>();
```

Comment:
There a reason we do this here rather than in `bump_warning_count` or along with the `WarningHandler` check???

My initial impression when reading the code was that it would make the intent clearer to have all of the `WarningHandler` related code together.  This might also remove the need for the other `remove` -> `get` change.

Reply:
> This might also remove the need for the other remove -> get change.

The way the code is written, it makes it look like the `remove` is intentional but I'm not seeing why.  If it is intentional, then this is a bug not uncovered by tests.  If its not intentional, than we should just use `get` because it would make the intent clearer.

I guess a third option is that its defensive design to avoid printing twice in case of a change.  Unsure how much we need that since we probably would want to know and fix it.

Reply:
I moved it into `report_warning_count`